### ISSUE TITLE
FP16 Support

### ISSUE THREAD

**Issue by zehuichen123** [joe-siyuan-qiao/DetectoRS#9]
Hi! Since DetectoRS is quite large, are there any plans to support fp16 training since I was not allowed to add other features to it due to 11GB memory limitation

**Comment by zehuichen123** [comment]
simply adding `fp16 = dict(loss_scale=512.)` yields Nan loss on my dataset :(

**Comment by joe-siyuan-qiao** [comment]
@zehuichen123 thanks for the question.

We haven't tried FP16 and to be honest, I'm not familiar with it either. If memory is the bottleneck, here's what I would do: 1. use resnet50. 2. change the batch size to 1 per GPU and adjust the learning rate. 3. set with_cp in the backbone to save memory. The 3rd one is quite useful if you haven't tried it already.

**Comment by zehuichen123** [comment]
Thanks, I'll give it a try!

**Comment by mayaya1** [comment]
> simply adding `fp16 = dict(loss_scale=512.)` yields Nan loss on my dataset :(

me too！did you solve this？
